packages/@expo/fingerprint/src/Fingerprint.ts

```diff
@@ -46,15 +51,56 @@ export async function diffFingerprintChangesAsync(
 }
 
 /**
- * Differentiate two fingerprints
+ * Differentiate two fingerprints with operation type.
+ * The implementation is assumed that the sources are sorted.
+ */
+export function diffFingerprintsWithOp(
+  fingerprint1: Fingerprint,
+  fingerprint2: Fingerprint
+): FingerprintDiffItem[] {
+  let index1 = 0;
+  let index2 = 0;
+  const diff: FingerprintDiffItem[] = [];
+
+  while (index1 < fingerprint1.sources.length && index2 < fingerprint2.sources.length) {
+    const source1 = fingerprint1.sources[index1];
+    const source2 = fingerprint2.sources[index2];
+
+    const compareResult = compareSource(source1, source2);
+    if (compareResult === 0) {
+      if (source1.hash !== source2.hash) {
+        diff.push({ op: 'changed', source: source2 });
+      }
+      ++index1;
+      ++index2;
+    } else if (compareResult < 0) {
+      diff.push({ op: 'removed', source: source1 });
+      ++index1;
+    } else {
+      diff.push({ op: 'added', source: source2 });
+      ++index2;
+    }
+  }
+
+  while (index1 < fingerprint1.sources.length) {
+    diff.push({ op: 'removed', source: fingerprint1.sources[index1] });
+    ++index1;
+  }
+  while (index2 < fingerprint2.sources.length) {
+    diff.push({ op: 'added', source: fingerprint2.sources[index2] });
+    ++index2;
+  }
+
+  return diff;
+}
+
+/**
+ * Differentiate two fingerprints and returns the changed sources.
+ * We keep this for backward compatibility.
```

Comment:
Given that fingerprint is still in beta, do we need to keep this for backward compatibility?

Reply:
tbh it's not necessary. i can change `diffFingerprints` to return in the new format. thanks!